aws/sns/resources.py

```diff
@@ -21,20 +22,49 @@ def sns_topics():
     )
 
 
-def sns_subscription_attributes():
+def sns_parent_topic_exists(subscription):
     "https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sns.html#subscription"
-    return [
+    topicArn = ":".join(
+        re.split(":", subscription)[0:6]
+    )  # Naive method of divining topic from subscription
```

Comment:
I'm not sure the format is specified and official part of the interface, but I wouldn't really expect this to change, so what you have seems fine. You could use `subscription.split(":")` (using the `str` method `split()`, which splits on literal strings rather than regular expressions).

If I run `aws sns list-subscriptions` from the command line, I get a list that includes both the subscription ARN and the topic ARN for each subscription. Maybe we can pass the whole JSON object (i.e. Python dict) to this function, rather than just the subscription ARN?

Reply:
My objection was primarily to using a split, slice, join operation to lop the end off a string.  Using a string method instead of a a regular expression is certainly an improvement.  I wasn't aware that the `list-subscriptions` returned the topic ARN.  That's certainly the way to go here.

Reply:
Agreed, if possible it's best to use the topic ARN returned by the API.